files/list.go

```diff
@@ -184,6 +189,24 @@ func shouldIncludeEntry(listPath, filePath string, entry fs.DirEntry, opts *List
 		if opts.Flags&Directories == 0 {
 			return false
 		}
+
+		// this is a directory and we are including directories
+		// check if we can include empty directories
+		if opts.Flags&NotEmpty != 0 {
+			ls, err := os.ReadDir(filePath)
+			if err != nil {
+				// if we can't read the contents
```

Comment:
Let's try to avoid needlessly multiline comments
```
// if dir read fails, skip this directory since the listing code will probably also fail to read it
```